### PR TITLE
Create MVC log viewer infrastructure

### PR DESCRIPTION
**Pull request by MichaelDeciso**
This should allow for the elimination of almost 30 old-style pages. For now, I have only ported syslog, configd.log, lighttpd.log and ppps.log.

### CODE REVIEW

File: src/opnsense/scripts/diagnostics/deletelog.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/local/bin/php
+<?php
```
Comment (fichtner) [inline src/opnsense/scripts/diagnostics/deletelog.php]: license missing

File: src/opnsense/scripts/diagnostics/viewlog.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#!/usr/local/bin/php
+<?php
```
Comment (fichtner) [inline src/opnsense/scripts/diagnostics/viewlog.php]: license missing

**Comment by AdSchellevis** [comment]
not going to add this now, doesn't really bring something new to the table.

File: src/opnsense/scripts/diagnostics/viewlog.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+echo json_encode($result) . PHP_EOL;
+exit(0);
+
+function doFileChecks($logfile)
+{
+    if (is_dir($logfile)) {
+        echo json_encode(['status' => sprintf(gettext('File %s is a directory.'), $logfile)]) . PHP_EOL;
+        exit(1);
+    }
+    if (!file_exists($logfile)) {
```
Comment (fabianfrz) [inline src/opnsense/scripts/diagnostics/viewlog.php]: is_readable

File: src/opnsense/scripts/diagnostics/viewlog.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+if ($type == 'clog') {
+    exec("/usr/local/sbin/clog " . escapeshellarg($logfile) . " {$grepline}| grep -v \"CLOG\" | grep -v \"\033\" | /usr/bin/tail {$tailParam} -n " . escapeshellarg($numlines), $logarr);
+} else {
+    exec("cat " . escapeshellarg($logfile) . " {$grepline} | /usr/bin/tail {$tailParam} -n " . escapeshellarg($numlines), $logarr);
+}
+
+$result = checkResult($logarr, $logfile);
+
+echo json_encode($result) . PHP_EOL;
```
Comment (fabianfrz) [inline src/opnsense/scripts/diagnostics/viewlog.php]: the eol is not needed, isn't it?

**Comment by fabianfrz** [comment]
@AdSchellevis it brings at least privilege separation. Otherwise, @MichaelDeciso may take a look at mine under https://github.com/opnsense/plugins/blob/master/www/nginx/src/opnsense/www/js/nginx/src/logviewer.js, which is a more powerful log viewer on client side.

**Comment by AdSchellevis** [comment]
@fabianfrz we're not going to do this now, it could be simplified (probably a lot), but costs us all valuable time at the moment. We could probably handle this all with only a couple of controllers, reducing our footprint a lot.

**Pull request closed without merging**